### ISSUE TITLE
Encrypted appservices checklist

### ISSUE THREAD

**Issue by turt2live** [turt2live/matrix-bot-sdk#135]
* [x] All of https://github.com/turt2live/matrix-bot-sdk/issues/134
* [ ] Synapse implementation of https://github.com/matrix-org/matrix-doc/pull/3202
* [ ] Wire up appservice side of SDK to the MatrixClient/CryptoClient bits

**Comment by turt2live** [comment]
Synapse side blocked on https://github.com/matrix-org/synapse/issues/10653

**Comment by turt2live** [comment]
Closing in favour of more detailed individual issues opened on the repo.
